src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1721 @@ const ActorInit Boss_Dodongo_InitVars = {
+            sp84.y = -1523.76f;
+            sp84.z = (cosf(temp_f10) * temp_f12) + (-3304.0f);
+            EffectSsGMagma2_Spawn(globalCtx, &sp84, &magmaPrimColor[magma2DrawMode], &magmaEnvColor[magma2DrawMode],
+                                  10 - (magma2DrawMode * 5), magma2DrawMode, magmaScale + 100);
+        }
+
+        if ((this->unk_19E & sp90) == 0) {
+            Vec3f sp6C = { 0.0f, 0.0f, 0.0f };
+            Vec3f sp60 = { 0.0f, 0.0f, 0.0f };
+            Vec3f sp54;
+            f32 sp50 = Rand_ZeroOne() * 330.0f;
+            f32 sp4C = Rand_ZeroOne() * 6.28f;
+
+            sp54.x = sinf(sp4C) * sp50 + (-890.0f);
+            sp54.y = -1523.76f;
+            sp54.z = (cosf(sp4C) * sp50) + (-3304.0f);
+            EffectSsGMagma_Spawn(globalCtx, &sp54);
+            for (i1 = 0; i1 < 4; i1++) {
+                sp60.y = 0.4f;
+                sp60.x = Rand_CenteredFloat(0.5f);
+                sp60.z = Rand_CenteredFloat(0.5f);
+                sp50 = Rand_ZeroOne() * 330.0f;
+                sp4C = Rand_ZeroOne() * 6.28f;
+                sp54.x = (sinf(sp4C) * sp50) + (-890.0f);
+                sp54.y = -1513.76f;
+                sp54.z = (cosf(sp4C) * sp50) + (-3304.0f);
+                func_808C17C8(globalCtx, &sp54, &sp6C, &sp60, ((s16)Rand_ZeroFloat(2.0f)) + 6, 0x50);
+            }
+        }
+
+        func_808C1554(&D_030021D8, D_808C73C8, this->unk_19E, this->unk_224);
+    }
+
+    if (this->unk_1C6 != 0) {
+        u16* ptr1 = SEGMENTED_TO_VIRTUAL(D_808C73C8);
+        u16* ptr2 = SEGMENTED_TO_VIRTUAL(D_808C93C8);
+        s16 i2;
+
+        for (i2 = 0; i2  < 20; i2++) {
+            s16 new_var = this->unk_1C2 & 0x7FF;
+
+            ptr1[new_var] = ptr2[new_var];
+            this->unk_1C2 += 37;
+        }
+        Math_SmoothStepToF(&this->unk_224, 0.0f, 1.0f, 0.01f, 0.0f);
+    }
+
+    if (this->unk_1BC == 0) {
+        if (this->actionFunc != BossDodongo_DeathCutscene) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+
+        if (this->actionFunc == BossDodongo_Roll) {
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+
+    this->collider.elements[0].dim.scale = (this->actionFunc == BossDodongo_Inhale) ? 0.0f : 1.0f;
+
+    for (i1 = 6; i1 < 19; i1++) {
+        if (i1 != 12) {
+            this->collider.elements[i1].dim.scale = (this->actionFunc == BossDodongo_Roll) ? 0.0f : 1.0f;
+        }
+    }
+
+    if (this->unk_244 != 0) {
+        gGameInfo->data[((5 * 6) * 16) + 64] = 1;
+        gGameInfo->data[((5 * 6) * 16) + 65] = 255;
+        gGameInfo->data[((5 * 6) * 16) + 66] = 80;
+        gGameInfo->data[((5 * 6) * 16) + 67] = 0;
+        gGameInfo->data[((5 * 6) * 16) + 68] = (u8)this->unk_244;
+    } else {
+        gGameInfo->data[((5 * 6) * 16) + 64] = 0;
```

Comment:
```suggestion
        MREG(64) = 1;
        MREG(65) = 255;
        MREG(66) = 80;
        MREG(67) = 0;
        MREG(68) = (u8)this->unk_244;
    } else {
        MREG(64) = 0;
```
There's a chance i might have calculated the REG macro wrong